### PR TITLE
feat: Add minimal hardhat setup

### PR DESCRIPTION
**Pull request by phated**
This ports over a lot of the Dark Forest hardhat setup into zk-game-starter. I also took the chance to improve the code-comment documentation on various config.

I set up two workspace packages that different plugins write to, but didn't fill out their entire package.json files. This will need to be done later.

I also reworked the workspace loading using `@npmcli/map-workspaces` which is **waayyy** better than my old way of looking up workspace packages in DF.

I explicitly left out:
1. hardhat-circom because I want to include the circuits and stuff when I add it
2. any contracts because I want to add the diamond pattern stuff as the first smart contract commit and I need to re-review the ecosystem first

cc @nalinbhardwaj I invited you to the project and I'd love a review!

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  defaultNetwork: "hardhat",
+  /**
+   * This configures which networks hardhat will communicate with
+   */
+  networks: {
+    localhost: {
+      url: "http://localhost:8545/",
+      accounts: {
+        mnemonic:
+          "change typical hire slam amateur loan grid fix drama electric seed label",
```
Comment (nalinbhardwaj) [inline hardhat.config.ts:34]: thoughts on moving these to variables in a `.env`/`.env.example` file?

I see exgrasia (so presumably some old version of DF also) used to do that https://github.com/nalinbhardwaj/exgrasia/blob/master/eth/hardhat.config.ts#L20

**Review by nalinbhardwaj — APPROVED** [review]
nice starting point! :))

im guessing other packages (client and networking stuff in particular?) are also coming later

File: hardhat.config.ts (hunk 1 above)
Comment (phated) [reply hardhat.config.ts:34]: Yep! DF still does this: https://github.com/darkforest-eth/eth/blob/master/hardhat.config.ts#L35-L37 and even determines which networks to add based on the deployer being available: https://github.com/darkforest-eth/eth/blob/master/hardhat.config.ts#L91-L95

One thing that I'd like to solve is supporting a private key or mnemonic before we add support for that. Currently this only has local networks, so I figured I could defer those improvements until later.

**Comment by phated** [comment]
Thanks for the review! Yeah, client, networking, etc are coming in separate PRs, similar to the contracts and circuits.

**Pull request merged**